x-pack/plugins/enterprise_search/public/applications/analytics/components/add_analytics_collections/add_analytics_collection_logic.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { i18n } from '@kbn/i18n';
+
+import { AnalyticsCollection } from '../../../../../common/types/analytics';
+import { HttpError, Status } from '../../../../../common/types/api';
+
+import { isAlphaNumericOrUnderscore } from '../../../../../common/utils/is_alphanumeric_underscore';
+import { generateEncodedPath } from '../../../shared/encode_path_params';
+import { flashAPIErrors, flashSuccessToast } from '../../../shared/flash_messages';
+import { KibanaLogic } from '../../../shared/kibana';
+import { AddAnalyticsCollectionsAPILogic } from '../../api/add_analytics_collection/add_analytics_collection_api_logic';
+import { COLLECTION_VIEW_PATH } from '../../routes';
+
+export interface AddAnalyticsCollectionsActions {
+  apiError(error: HttpError): HttpError;
```

Comment:
is there a better way I can do this without c&p? Thinking out loud - Could we have a base connect interface that we extend from? `interface ConnectActions<T> { apiError(error: HttpError) : HttpError }` for example. The challenge here is when these methods are renamed.

Reply:
We generally grab them from the constructed type in the API logic, rather than typing them manually. We have an Actions type to help you there. In the `new_search_index_logic` we have this, for instance:
```
type NewSearchIndexActions = Pick<
  Actions<IndexExistsApiParams, IndexExistsApiResponse>,
  'makeRequest'
> & {
  apiIndexCreated: Actions<
    CreateApiIndexApiLogicArgs,
    CreateApiIndexApiLogicResponse
  >['apiSuccess'];
  connectorIndexCreated: Actions<
    AddConnectorApiLogicArgs,
    AddConnectorApiLogicResponse
  >['apiSuccess'];
  crawlerIndexCreated: Actions<CreateCrawlerIndexArgs, CreateCrawlerIndexResponse>['apiSuccess'];
};
```

Because we often need to rename, I don't think another helper type would make a ton of sense tbh.

Reply:
Thanks! I will use this way